File: Data Structures & Algorithms/contains-duplicate-ii/submission-7.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {boolean}
     */
    containsNearbyDuplicate(nums: number[], k: number): boolean {
        let window: Set<number> = new Set();
        let l = 0;
        for (let r = 0; r < nums.length; r++ ) {
            if (r - l > k) {
                window.delete(nums[l]);
                l++;
            }
            if (window.has(nums[r])){
                return true;
            }
            window.add(nums[r]);
        }
        return false;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-7.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        f1, f2 = {}, {}
        for c in s:
            f1[c] = f1.get(c, 0) + 1
        
        for c in t:
            f2[c] = f2.get(c, 0) + 1
        
        return f1 == f2